src/general/storageclass/src.rego

```diff
@@ -0,0 +1,48 @@
+package k8sstorageclass
+
+is_pvc(obj) {
+  obj.apiVersion == "v1"
+  obj.kind == "PersistentVolumeClaim"
+}
+
+is_statefulset(obj) {
+  obj.apiVersion == "apps/v1"
+  obj.kind == "StatefulSet"
+}
+
+violation[{"msg": msg}] {
+  count(data.inventory.cluster["storage.k8s.io/v1"]["StorageClass"]) == 0
+  msg := sprintf("storageclasses not synced. Gatekeeper is misconfigured.", [])
+}
+
+storageclass_found(name) {
+  data.inventory.cluster["storage.k8s.io/v1"]["StorageClass"][name]
+}
+
+violation[{"msg": msg}] {
+  is_pvc(input.review.object)
+  not storageclass_found(input.review.object.spec.storageClassName)
+  msg := sprintf("pvc did not specify a valid storage class name <%v>. Must be one of [%v]", [input.review.object.spec.storageClassName, concat(", ", [n | data.inventory.cluster["storage.k8s.io/v1"]["StorageClass"][n]])])
```

Comment:
Could there be privacy and security concerns sharing this info since this is printing out information of all storageclass, a cluster-level resource with a user that may not have cluster-level access rbac?

Reply:
I debated that... on my cluster, I don't allow them to be listed, just because the details could be listed too. but in this case, its just the names. so, I don't know.... I guess we could maybe add a parameter to allow this to be configured?

Reply:
Made it conditional